src/dmd/root/filemanager.d

```diff
@@ -0,0 +1,121 @@
+/**
+ * Read a file from disk and store it in memory.
+ *
+ * Copyright: Copyright (C) 1999-2020 by The D Language Foundation, All Rights Reserved
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:    $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/root/filemanager.d, root/_filemanager.d)
+ * Documentation:  https://dlang.org/phobos/dmd_root_filemanager.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/root/filemanager.d
+ */
+
+module dmd.root.filemanager;
+
+import dmd.root.stringtable : StringTable;
+import dmd.root.file : File, FileBuffer;
+import dmd.root.filename : FileName;
+import dmd.dmodule : getFilename, lookForSourceFile;
+import dmd.identifier;
+import dmd.arraytypes;
+
+struct FileManager
+{
+    private StringTable!(FileBuffer*) files;
+
+    void _init()
+    {
+        files._init();
+    }
+
+    /**
+     * Looks up the given filename in memory.
+     *
+     * Returns: the loaded source file if it was found in memory,
+     *      otherwise `null`
+     */
+    FileBuffer* opIndex(const(FileName) filename)
```

Comment:
This is `const` while the function below is not. `FileName` is a struct which is passed by value. Is there a reason for it to be `const`?